### DISCUSSION TITLE
Django Media Files

### DISCUSSION THREAD

**Discussion by daronzwink** [nickjj/docker-django-example#37] [Q&A] [answered]
I am attempting to configure support for uploading of files to Django, but not able to figure out what configuration changes are needed to support serving media files and making them available through the docker configuration (similar to how static files are available).

Has anyone had any luck adding the configuration needed for this?

**Comment by nickjj** [comment]
Hi,

What does your current configuration look like? Is there something specific to this project that's preventing it from working? The static directory is configured in `src/config/settings.py`, this would happen with or without Docker.

https://docs.djangoproject.com/en/4.2/topics/http/file-uploads/ has a few examples of uploading files.

**Comment by daronzwink** [comment, marked as answer]
I may have been overthinking the configuration and I now have media files working. My configuration updates are below, and hopefully they look correct and can help others out.

**src/config/settings.py**
```python
# Media files
# https://docs.djangoproject.com/en/4.2/topics/files/
MEDIA_URL = "/media/"
MEDIA_ROOT = os.path.join(BASE_DIR, "media_collected")
```

**src/config/urls.py**
```python
from django.conf import settings
from django.conf.urls.static import static
...
if settings.DEBUG:
    urlpatterns += static(settings.MEDIA_URL, document_root=settings.MEDIA_ROOT)
```

**.gitignore**
```gitignore
# Media files
media_collected/
```
